internal/report/report.go

```diff
@@ -175,6 +189,13 @@ func (writer Writer) isExcluded(v report.Vulnerability, target string) (bool, er
 				continue
 			}
 		}
+
+		if !excl.ExpirationDate.IsZero() {
+			now := writer.timeProvider.Now()
+			if excl.ExpirationDate.Before(now) {
+				continue
+			}
+		}
```

Comment:
I move this check to the beginning and if the exclusion is expired, then just `continue` because it must be ignored. So, it does not make sense to check anything else. Also, after my previous comment, the code would become

```suggestion
		if !excl.ExpirationDate.IsZero() && excl.ExpirationDate.Before(timeNow) {
			continue
		}
```